src/local.c

```diff
@@ -140,87 +140,102 @@ int
         return 1;
-
-    outlen = TMP_DIR_LEN;
-    if (sstrncat(outfile, &outlen, STRBUFSIZ, TMP_FILE, TMP_FILE_LEN))
+    }
+    if (sstrncat(tldr_home, &outlen, STRBUFSIZ, TLDR_HOME, TLDR_HOME_LEN)) {
         return 1;
-
-    if (download_file(ZIP_URL, outfile, verbose)) {
-        fprintf(stderr, "Error: Downloading File: %s\n", ZIP_URL);
+    }
+    if (mkdir(tldr_home, 0755) > 0 && errno != EEXIST) {
+        fprintf(stderr, "Error: Could not create directory: %s\n", tldr_home);
         return 1;
     }
 
-    if (unzip(outfile, outpath)) {
-        rm(outpath, 0);
+    /* Set up the temp directory */
+    outlen = 0;
+    if (sstrncat(temp_dir, &outlen, STRBUFSIZ, tldr_home, strlen(tldr_home))) {
+        return 1;
+    }
+    if (sstrncat(temp_dir, &outlen, STRBUFSIZ, TMP_DIR, TMP_DIR_LEN)) {
+        return 1;
+    }
+    if (mkdir(temp_dir, 0755) > 0 && errno != EEXIST) {
+        fprintf(stderr, "Error: Could not create directory: %s\n", temp_dir);
         return 1;
     }
 
+    /* The update zip file will be downloaded to $HOME/TLDR_HOME/TMP_DIR/TMP_FILE */
     outlen = 0;
-    if (sstrncat(tmp, &outlen, STRBUFSIZ, outpath, strlen(outpath)))
+    if (sstrncat(update_zip_file, &outlen, STRBUFSIZ, temp_dir, strlen(temp_dir))) {
         return 1;
-    if (sstrncat(tmp, &outlen, STRBUFSIZ, TLDR_ZIP_DIR, TLDR_ZIP_DIR_LEN))
+    }
+    if (sstrncat(update_zip_file, &outlen, STRBUFSIZ, TMP_FILE, TMP_FILE_LEN)) {
         return 1;
+    }
 
-    homedir = gethome();
-    if (homedir == NULL) {
-        fprintf(stderr, "Error: HOME not existant\n");
+    /* Download and unzip the file */
+    if (download_file(ZIP_URL, update_zip_file, verbose)) {
+        fprintf(stderr, "Error: Downloading file: %s\n", ZIP_URL);
+        return 1;
+    }
+
+    if (unzip(update_zip_file, temp_dir)) {
+        rm(temp_dir, 0);
         return 1;
     }
 
     outlen = 0;
-    if (sstrncat(outhome, &outlen, STRBUFSIZ, homedir, strlen(homedir))) {
+    if (sstrncat(update_contents, &outlen, STRBUFSIZ, temp_dir, strlen(temp_dir))) {
         return 1;
     }
-    if (sstrncat(outhome, &outlen, STRBUFSIZ, TLDR_HOME, TLDR_HOME_LEN)) {
+    if (sstrncat(update_contents, &outlen, STRBUFSIZ, TLDR_ZIP_DIR, TLDR_ZIP_DIR_LEN)) {
         return 1;
     }
 
-    if (mkdir(outhome, 0755) > 0 && errno != EEXIST) {
-        fprintf(stderr, "Error: Could Not Create Directory: %s\n", outhome);
-        rm(outpath, 0);
+    /* tldr_home_db is where we want to move the update contents */
+    outlen = 0;
+    if (sstrncat(tldr_home_db, &outlen, STRBUFSIZ, tldr_home, strlen(tldr_home))) {
         return 1;
     }
-
-    if (sstrncat(outhome, &outlen, STRBUFSIZ, TLDR_DIR, TLDR_DIR_LEN))
+    if (sstrncat(tldr_home_db, &outlen, STRBUFSIZ, TLDR_DIR, TLDR_HOME_LEN)) {
         return 1;
-    if (sstrncat(outhome, &outlen, STRBUFSIZ, "/", 1))
+    }
+    if (sstrncat(tldr_home_db, &outlen, STRBUFSIZ, "/", 1)) {
```

Comment:
I don't think the trailing slash is necessary. Can you confirm?
If not, then it should be safe to remove these 3 lines.

I've tested your branch without them, and it appears to work fine.

Maybe it'd cause issues if we had a file named `tldr` in the same directory, but that's not the case. Given that the state of this directory is managed by us, I don't think it's a concern.